src/modules/keyboardmanager/KeyboardManagerEngineLibrary/KeyboardEventHandlers.cpp

```diff
@@ -579,46 +579,90 @@ namespace KeyboardEventHandlers
                             }
 
                             size_t key_count;
-                            LPINPUT keyEventList;
+                            LPINPUT keyEventList = nullptr;
+                            
+                            // Check if a new remapping should be applied
+                            Shortcut currentlyPressed = it->first;
+                            currentlyPressed.actionKey = data->lParam->vkCode;
+                            if (reMap.find(currentlyPressed) != reMap.end())
+                            {
+                                auto& newRemapping = reMap[currentlyPressed];
+                                Shortcut from = std::get<Shortcut>(it->second.targetShortcut);
+                                if (newRemapping.RemapToKey())
+                                {
+                                    DWORD to = std::get<0>(newRemapping.targetShortcut);
+                                    key_count = from.Size() - 1 + 1;
+                                    keyEventList = new INPUT[key_count]();
+                                    memset(keyEventList, 0, sizeof(keyEventList));
```

Comment:
I [believe we already zero-initialize](https://stackoverflow.com/a/7546745/657390) the list